libs/pageserver_api/src/key.rs

```diff
@@ -141,6 +141,7 @@ impl Key {
     }
 }
 
+#[inline(always)]
```

Comment:
Was the compiler failing to inline this automatically?

Reply:
Didn't check, but, we definitely want it to happen here.

Reply:
> Non-generic functions will never be inlined across crate boundaries unless link-time optimization is used; generic functions might be.

https://nnethercote.github.io/perf-book/inlining.html

Reply:
Wondering what other functions in `pageserver_api` we want to mark as inlined 🤔